lib/relationships.js

```diff
@@ -0,0 +1,96 @@
+var _ = require('lodash');
+var inflection = require('inflection');
+
+module.exports = function (app, options) {
+  //get remote methods.
+  //set strong-remoting for more information
+  //https://github.com/strongloop/strong-remoting
+  var remotes = app.remotes();
+
+  remotes.before('**', function (ctx, next) {
+
+    var id = ctx.req.params.id;
+    var data = ctx.args.data;
+
+    relationships(id, data, app);
+
+    next();
+  });
+
+  // for create
+  remotes.after('**', function (ctx, next) {
+
+    if (!ctx.result) {
+      next();
+    };
+    var id = ctx.result.id;
+    var data = ctx.req.body;
+
+    relationships(id, data, app);
+
+    next();
+  });
+
+};
+
+function relationships (id, data, app) {
+
+  if (!data || !data.data || !data.data.type || !id) {
+    return;
+  }
+
+  var modelName = inflection.classify(data.data.type);
+
+  var model = app.models[modelName];
+  if (!model || !data.relationships) {
+    return;
+  }
+
+  _.each(data.relationships, function (relationship, name) {
+
+    var serverRelation = model.relations[name];
+    var modelTo = serverRelation.modelTo;
+    if (!modelTo) {
+      return false;
+    }
+    var fk = serverRelation.keyTo;
+
+    var setTo = {};
+    setTo[fk] = null;
+    var where = {};
+    where[fk] = id;
+
+    // remove all relations to the model (eg .: post/1)
+    modelTo.updateAll(where, setTo, function (err, info) {
+      if (err) {console.log(err);}
+    });
+
+    if (_.isArray(relationship.data)) {
+      // find all instance from the relation data eg
+      // [{type: "comments", id: 1}, {type: "comments", id: 2}]
+      _.each(relationship.data, function (item) {
+        createOrUpdateRelation(modelTo, item.id, where);
+      });
+    } else {
+      // relationship: {data: {type: "comments": id: 1}}
+      createOrUpdateRelation(modelTo, relationship.data.id, where);
```

Comment:
I'm actually a little unsure if we should be doing `creates` at all. The relationship data that comes from the client only ever provides: {type:'models', id:1} so while we can create a new related model from that, we have no data other than the foreign key and id to set. Might need to go back to the spec and check but it seems to me like maybe this functionality is only for linkage/unlinkage between models. I had assumed you would need to create the other model first and then link them.

Eg. 
POST /api/comments
```js
{
  data: {
    type: 'comments'
    attributes: {
      name: 'my comment',
      comment: 'my comment text'
    }
  }
}
```
That would give you an id (eg. 1) back for the created comment which you could then set on you client side post model before calling save which would send:

POST /api/posts
```js
{
  data: {
    type: 'posts'
    attributes: {
      name: 'my post',
      comment: 'my comment text'
    },
    relationships: {
      comments: {
        {type: 'comments', id: 1}
      }
    }
  }
}
```
I'll have another look at the spec.
Thoughts?